### PR TITLE
fix(traffix-split): multiple rules with multiple weighted_upstreams under each rule cause upstream_key duplicate

### PR DESCRIPTION
**Pull request by tzssangglass**
Signed-off-by: tzssangglass <tzssangglass@gmail.com>

### What this PR does / why we need it:

fix: #5276
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/traffic-split.lua [hunk 1]
```diff
@@ -203,6 +207,14 @@ local function new_rr_obj(weighted_upstreams)
         elseif upstream_obj.upstream then
             -- Add a virtual id field to uniquely identify the upstream key.
             upstream_obj.upstream.vid = i
+            if upstream_obj.upstream.nodes then
```
Comment (spacewander) [inline apisix/plugins/traffic-split.lua:210]: Look like the nodes check here can be skipped?

File: apisix/plugins/traffic-split.lua [hunk 2]
```diff
@@ -203,6 +207,14 @@ local function new_rr_obj(weighted_upstreams)
         elseif upstream_obj.upstream then
             -- Add a virtual id field to uniquely identify the upstream key.
             upstream_obj.upstream.vid = i
```
Comment (spacewander) [inline apisix/plugins/traffic-split.lua:209]: It seems the vid is no longer required?

Comment (tzssangglass) [reply apisix/plugins/traffic-split.lua:209]: may not, here test vid:https://github.com/apache/apisix/blob/4dafab5afa3293b3d72007517246e01da385f8ef/t/plugin/traffic-split3.t#L398-L461

File: apisix/plugins/traffic-split.lua (hunk 1 above)
Comment (tzssangglass) [reply apisix/plugins/traffic-split.lua]: remove it

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Review by shuaijinchao — APPROVED** [review]

**Pull request merged**